geth/proxy/transactions/sendtransaction.go

```diff
@@ -0,0 +1,315 @@
+	}
+
+	// Attempt to get the hex version of the transaction.
+	txBytes, err := rlp.EncodeToBytes(txs)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx2, canceler2 := context.WithDeadline(context.Background(), time.Now().Add(1*time.Minute))
+	defer canceler2()
+
+	var result json.RawMessage
+	if err := client.CallContext(ctx2, &result, "eth_sendRawTransaction", gethcommon.ToHex(txBytes)); err != nil {
+		return nil, err
+	}
+
+	resp.Set("result", result)
+	resp.Set("hash", txs.Hash().String())
+
+	return resp, nil
+}
+
+// ExecuteSendTransaction defines a function which handles the procedure called for the dealing with
+// RPCCalls with "eth_sendTransaction" Methods.
+func ExecuteSendTransaction(manager common.NodeManager, req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
+	resp, _ := call.Otto.Object(`({"jsonrpc":"2.0"})`)
+	resp.Set("id", req.ID)
+
+	txHash, err := processRPCCall(manager, req, call)
+	resp.Set("result", txHash.Hex())
+
+	if err != nil {
+		resp = newErrorResponse(call, -32603, err.Error(), &req.ID).Object()
+		return resp, nil
+	}
+
+	return resp, nil
+}
+
+func processRPCCall(manager common.NodeManager, req common.RPCCall, call otto.FunctionCall) (gethcommon.Hash, error) {
+	lightEthereum, err := manager.LightEthereumService()
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	backend := lightEthereum.StatusBackend
+
+	messageID, err := preProcessRequest(call.Otto, req)
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	// onSendTransactionRequest() will use context to obtain and release ticket
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, common.MessageIDKey, messageID)
+
+	//  this call blocks, up until Complete Transaction is called
+	txHash, err := backend.SendTransaction(ctx, sendTxArgsFromRPCCall(req))
+	if err != nil {
+		return gethcommon.Hash{}, err
+	}
+
+	// invoke post processing
+	postProcessRequest(call.Otto, req, messageID)
+
+	return txHash, nil
+}
+
+//==========================================================================================================
+
+// ExecuteOtherTransaction defines a function which handles the processing of non `eth_sendTransaction`
+// requests. It is expected that this method does not require signing the transaction.
+func ExecuteOtherTransaction(manager common.NodeManager, req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
+	client, err := manager.RPCClient()
+	if err != nil {
+		return nil, common.StopRPCCallError{Err: err}
+	}
+
+	JSON, _ := call.Otto.Object("JSON")
+
+	var result json.RawMessage
+
+	resp, _ := call.Otto.Object(`({"jsonrpc":"2.0"})`)
+	resp.Set("id", req.ID)
+
+	// do extra request pre processing (persist message id)
+	// within function semaphore will be acquired and released,
+	// so that no more than one client (per cell) can enter
+	messageID, err := preProcessRequest(call.Otto, req)
+	if err != nil {
+		return nil, common.StopRPCCallError{Err: err}
+	}
+
+	errc := make(chan error, 1)
```

Comment:
I think we need a buffered channel because nothing reads from `errc` while `client.Call()` writes to it. It will be a deadlock. We could do something like:
```go
errc := make(chan error)
go func() {
  errc <- client.Call(&result, req.Method, req.Params...)
}()
err := <-errc 
```
but I think a buffered channel is equivalent solution.